### ISSUE TITLE
Restarting server resets position of goal window

### ISSUE THREAD

**Issue by Alizter** [ejgallego/coq-lsp#678]
I'm not sure if this is a VSCode bug or something we can fix in the coq-lsp.

When you restart the coq-lsp server, the goals window opens and closes which means that it ends up getting grouped with your tab group rather than being on the side.

Here is before restarting:

![image](https://github.com/ejgallego/coq-lsp/assets/8614547/24ac841b-6c86-4ffc-b218-31dae62dc62a)

here is after restarting:

![image](https://github.com/ejgallego/coq-lsp/assets/8614547/57ad6364-9072-43b6-bc64-06c93b99793d)

It would be nice if we didn't have to manually fix this every time we restart.

Of course, another way to fix this is not to have to restart so often, but this issue is still somewhat annoying.

**Comment by ejgallego** [comment]
I cannot reproduce in vscode 1.90 , is that still a problem for you @Alizter ?

I think this may have been fixed by #755 and related PRs.

**Comment by Alizter** [comment]
Yes this appears to be fixed, but not by #755 IIUC. On my version there is a race that #755 is supposed to fix I see.

There is a related issue where you have 3 buffers horizontally and you restart, it turns into 2 and then opens a new tab for the goals on the right buffer rather than returning the 3 from before.
